pkg/guacrest/openapi.yaml

```diff
@@ -99,7 +99,30 @@ paths:
           $ref: "#/components/responses/InternalServerError"
         "502":
           $ref: "#/components/responses/BadGateway"
-
+  "/query/latest-sbom":
+    get:
+      summary: Retrieve the latest SBOM for a given package
+      operationId: findLatestSBOM
+      parameters:
```

Comment:
I'm not sure if I agree that searching by id is the best choice here:
- it requires a previous call to the GQL API (and an understanding of the package trie) which is partly what the REST API intends to avoid.
- I think it makes sense to be consistent across the REST API, and the other endpoint doesn't expose the internal identifiers

I also don't think that searching for SBOMs by package is the right approach:
- SBOMs are generated for and should be associated with artifacts, not packages (https://blog.deps.dev/zillions-of-sboms/). In some cases the data quality isn't there and they will be associated with packages, but I think it makes sense to have the interface work with the "right" way. 
- this interface hides many details -- does it mean the latest SBOM associated with the package, or the latest SBOM associated with the latest artifact occurrence of the package, or the latest among those two? 


IMO the best interface is with a digest. It is precise and subsumes searching for an SBOM by package.

Reply:
@mdeicas did you see the design issue: https://github.com/guacsec/guac/issues/2058?

Reply:
I did - I agree that it shouldn't be the role of these endpoints to "help you look for software", that the interface should identify the inputs precisely. But I guess I don't see what benefits there are to Guac ids in the interface. On the other hand, digests and purls (a purl as an identifier for a node in Guac), which are also precise, avoid exposing the internal identifiers.